src/TraceEvent/TraceLog.cs

```diff
@@ -7111,7 +7111,13 @@ internal TraceModuleFile UniversalMapping(ProcessMappingTraceData data, ProcessM
             // A loaded and managed modules depend on a module file, so get or create one.
             // TODO: We'll need to store FileOffset as well to handle elf images.
             TraceModuleFile moduleFile = process.Log.ModuleFiles.GetOrCreateModuleFile(data.FileName, data.StartAddress);
-            moduleFile.imageSize = (long)(data.EndAddress - data.StartAddress);
+            long newImageSize = (long)(data.EndAddress - data.StartAddress);
+            
+            // If we found an existing module file, check if we need to expand its size to handle overlapping mappings
+            if (moduleFile.imageSize < newImageSize)
+            {
+                moduleFile.imageSize = newImageSize;
+            }
 
             // Get or create the loaded module.
             TraceLoadedModule loadedModule = FindModuleAndIndexContainingAddress(data.StartAddress, data.TimeStampQPC, out index);
```

Comment:
`GetOrCreateModule` will only return a `TraceModuleFile` whose `ImageBase` matches the incoming mapping, so if the mapping happens to have the same name, as does the mappings for jitted code on Linux, then `GetOrCreateModule` will just create a new module.

`FindModuleAndIndexContainingAddress` will return a `TraceLoadedModule` that contains `data.StartAddress`, but doesn't necessarily start with it.  If it doesn't start with it, then we need to create a new `TraceLooadedModule` and insert it - this is needed to handle the overlapping mapping case.